[Bug]: Getting error building storybook in v10 with nextjs

### Describe the bug

I have a project that explicitly installs `next@14` rather than just implicitly installing it via `@storybook/next`
When I build the storybook I get the following error:

```
=> Failed to build the preview
SB_BUILDER-WEBPACK5_0002 (WebpackInvocationError): Module not found: Error: Can't resolve 'next/dist/server/request/draft-mode' in './node_modules/@storybook/nextjs/dist/export-mocks/headers'
    at file://./node_modules/@storybook/builder-webpack5/dist/index.js:1114:35
    at ./node_modules/webpack/lib/HookWebpackError.js:66:2
    at Hook.eval [as callAsync] (eval at create (./node_modules/tapable/lib/HookCodeFactory.js:31:10), <anonymous>:11:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (./node_modules/tapable/lib/Hook.js:21:14)
    at Cache.shutdown (./node_modules/webpack/lib/Cache.js:157:23)
    at ./node_modules/webpack/lib/Compiler.js:1407:15
    at Hook.eval [as callAsync] (eval at create (./node_modules/tapable/lib/HookCodeFactory.js:31:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (./node_modules/tapable/lib/Hook.js:21:14)
    at Compiler.close (./node_modules/webpack/lib/Compiler.js:1400:23)
    at file://./node_modules/@storybook/builder-webpack5/dist/index.js:1114:18
    at finalCallback (./node_modules/webpack/lib/Compiler.js:519:32)
    at onCompiled (./node_modules/webpack/lib/Compiler.js:533:20)
    at ./node_modules/webpack/lib/Compiler.js:1355:21
    at eval (eval at create (./node_modules/tapable/lib/HookCodeFactory.js:31:10), <anonymous>:32:1)
    at ./node_modules/webpack/lib/EntryPlugin.js:49:5
    at ./node_modules/webpack/lib/Compilation.js:2486:13
    at ./node_modules/webpack/lib/Compilation.js:2371:6
    at ./node_modules/webpack/lib/Compilation.js:2314:13
    at ./node_modules/webpack/lib/util/AsyncQueue.js:390:5
    at Hook.eval [as callAsync] (eval at create (./node_modules/tapable/lib/HookCodeFactory.js:31:10), <anonymous>:15:1)
    at AsyncQueue._handleResult (./node_modules/webpack/lib/util/AsyncQueue.js:360:21)
    at ./node_modules/webpack/lib/util/AsyncQueue.js:343:11
    at onTransitiveTasksFinished (./node_modules/webpack/lib/Compilation.js:1702:20)
    at ./node_modules/webpack/lib/Compilation.js:1688:7
    at ./node_modules/webpack/lib/Compilation.js:2314:13
    at ./node_modules/webpack/lib/util/AsyncQueue.js:390:5
    at Hook.eval [as callAsync] (eval at create (./node_modules/tapable/lib/HookCodeFactory.js:31:10), <anonymous>:15:1)
    at AsyncQueue._handleResult (./node_modules/webpack/lib/util/AsyncQueue.js:360:21)
    at ./node_modules/webpack/lib/util/AsyncQueue.js:343:11
    at onTransitiveTasksFinished (./node_modules/webpack/lib/Compilation.js:1702:20)
    at ./node_modules/webpack/lib/Compilation.js:1688:7
    at ./node_modules/webpack/lib/Compilation.js:2314:13
    at ./node_modules/webpack/lib/util/AsyncQueue.js:390:5
    at Hook.eval [as callAsync] (eval at create (./node_modules/tapable/lib/HookCodeFactory.js:31:10), <anonymous>:15:1)
    at AsyncQueue._handleResult (./node_modules/webpack/lib/util/AsyncQueue.js:360:21)
    at ./node_modules/webpack/lib/util/AsyncQueue.js:343:11
    at onTransitiveTasksFinished (./node_modules/webpack/lib/Compilation.js:1702:20)
    at ./node_modules/webpack/lib/Compilation.js:1688:7
    at ./node_modules/webpack/lib/Compilation.js:2314:13
    at ./node_modules/webpack/lib/util/AsyncQueue.js:390:5
    at Hook.eval [as callAsync] (eval at create (./node_modules/tapable/lib/HookCodeFactory.js:31:10), <anonymous>:15:1)
    at AsyncQueue._handleResult (./node_modules/webpack/lib/util/AsyncQueue.js:360:21)
    at ./node_modules/webpack/lib/util/AsyncQueue.js:343:11
    at onTransitiveTasksFinished (./node_modules/webpack/lib/Compilation.js:1702:20)
    at ./node_modules/webpack/lib/Compilation.js:1688:7
    at ./node_modules/webpack/lib/Compilation.js:2314:13
    at ./node_modules/webpack/lib/util/AsyncQueue.js:382:6
    at process.processTicksAndRejections (node:internal/process/task_queues:85:11)
```

Digging into directory structure I see that there is not a `next/dist/server/request/draft-mode` in the installed `next@14`. According to the next [documentation](https://nextjs.org/docs/app/api-reference/functions/draft-mode) draft mode is in `next/headers`.  This [line](https://github.com/storybookjs/storybook/blob/9d1fc3ebe4665f2b7ff5b30135504282e1296d04/code/frameworks/nextjs/src/export-mocks/headers/index.ts#L1) is importing the offending, non-existant file in my installation:

Not sure why this is happening, but it is preventing my storybook 10 upgrade. 

My `main.ts` looks like:
```
import type { StorybookConfig } from 'storybook/internal/types';

const config: StorybookConfig = {
  stories: ['../src/__stories__/**/*.stories.tsx'],
  addons: ['@storybook/addon-a11y', '@storybook/addon-docs'],
  framework: '@storybook/nextjs',
};

module.exports = config;
```

According to a comment in the original [PR](https://github.com/storybookjs/storybook/commit/59df579fbc5861357b99eae6578c77694b218224), 
```
// This export won't exist in Next.js 14 but it's safe because we ignore it in Webpack when applicable
```
But it seems not to be in my case

### Reproduction link

TBD

### Reproduction steps

_No response_

### System

```bash
Storybook Environment Info:

  System:
    OS: macOS 26.0.1
    CPU: (12) arm64 Apple M2 Pro
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.21.0 - ~/.nvm/versions/node/v22.21.0/bin/node
    npm: 10.9.4 - ~/.nvm/versions/node/v22.21.0/bin/npm <----- active
    pnpm: 10.1.0 - /opt/homebrew/bin/pnpm
  Browsers:
    Chrome: 142.0.7444.60
    Safari: 26.0.1
  npmPackages:
    @storybook/addon-a11y: 10.0.4 => 10.0.4 
    @storybook/addon-docs: 10.0.4 => 10.0.4 
    @storybook/nextjs: 10.0.4 => 10.0.4 
    storybook: 10.0.4 => 10.0.4
```

### Additional context

_No response_

--- a/code/frameworks/nextjs/src/export-mocks/headers/index.ts
+++ b/code/frameworks/nextjs/src/export-mocks/headers/index.ts
@@ -1,4 +1,3 @@
-import { draftMode as originalDraftMode } from 'next/dist/server/request/draft-mode';
 import * as headers from 'next/dist/server/request/headers';
 import { fn } from 'storybook/test';
 
@@ -10,5 +9,18 @@ export { headers } from './headers';
 export { cookies } from './cookies';
 
 // passthrough mocks - keep original implementation but allow for spying
-const draftMode = fn(originalDraftMode ?? (headers as any).draftMode).mockName('draftMode');
+// In Next.js 14, draftMode is exported from 'next/dist/client/components/headers'
+// In Next.js 15+, draftMode is exported from 'next/dist/server/request/draft-mode'
+// The webpack alias handles this, but we need to avoid the top-level import
+// to prevent build errors when the module doesn't exist
+let originalDraftMode: any;
+try {
+  // This will be resolved by webpack alias to the correct location based on Next.js version
+  originalDraftMode = require('next/dist/server/request/draft-mode').draftMode;
+} catch {
+  // Fallback to the location in the headers module (Next.js 14)
+  originalDraftMode = (headers as any).draftMode;
+}
+
+const draftMode = fn(originalDraftMode).mockName('draftMode');
 export { draftMode };

